### PR TITLE
Handle onSave after image is edited.

### PR DESCRIPTION
**Pull request by pszczesniak**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal(ckbox): Handling callback after edited image changes are saved. Closes cksource/ckeditor5-commercial#5635.

---

### Additional information

**TOKEN\_URL = '**[**https://preview.ckbox.io/cke-workspaces-demo/?page=token**](https://preview.ckbox.io/cke-workspaces-demo/?page=token)**'**

☝️ _This token endpoint should be used in manual test to test the CKBox Image Editor (getAsset wont throw any errors)._

### CODE REVIEW

File: packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts [hunk 1]
```diff
@@ -138,8 +148,34 @@ export default class CKBoxImageEditCommand extends Command {
...
+			} = preparedAsset.attributes as CKBoxAssetImageAttributesDefinition;
+
+			editor.model.change( writer => {
+				imageCommand.execute( {
+					source: {
+						src: imageFallbackUrl,
+						sources: imageSources,
+						alt: imageTextAlternative
+					}
+				} );
```
[the remark this answers is not in the document]
Comment (pszczesniak) [reply packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts:173]: @niegowski will update it after merging `master` with `epic` branch (and this one).

File: packages/ckeditor5-ckbox/src/ckboxcommand.ts [hunk 2]
```diff
@@ -343,7 +343,7 @@ export default class CKBoxCommand extends Command {
 /**
  * Parses the chosen assets into the internal data format. Filters out chosen assets that are not allowed.
  */
-function prepareAssets(
+export function prepareAssets(
```
Comment (arkflpc) [inline packages/ckeditor5-ckbox/src/ckboxcommand.ts:346]: I'd mark it as internal.

File: packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts [hunk 3]
```diff
@@ -92,7 +99,10 @@ export default class CKBoxImageEditCommand extends Command {
 		return {
 			tokenUrl: ckboxConfig.tokenUrl,
 			onClose: () => this.fire<CKBoxImageEditorEvent<'close'>>( 'ckboxImageEditor:close' ),
-			onSave: () => this.fire<CKBoxImageEditorEvent<'save'>>( 'ckboxImageEditor:save' )
+			// Timeout for demo purpose.
+			onSave: ( data: CKBoxRawAssetDefinition ) => setTimeout( () =>
```
Comment (arkflpc) [inline packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts:103]: I'm not sure if we want to delay this event. Let's discuss this. The possible options I see are:
1. Move the waiting (setTimeout for now) into `ckboxImageEditor:save` listener
2. Emit two events, one for save, one when the image has been processed.

We would have a ready point for placing the indicator.

**Review by arkflpc — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-ckbox/src/ckboxcommand.ts (hunk 2 above)
Comment (pszczesniak) [reply packages/ckeditor5-ckbox/src/ckboxcommand.ts:348]: Fixed in 4035000

File: packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pszczesniak) [reply packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts]: Added in 5147d24

File: packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts (hunk 3 above)
Comment (pszczesniak) [reply packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts:103]: Yes, i think that two events is a good idea, but `ckboxImageEditor:save`  especially the `save` word is not fit for me. Better would be have two events like:

*   `ckboxImageEditor:beginSaveProcess`,
*   `ckboxImageEditor:finishSaveProcess`

WDYT?

BTW - don't know how far is @illia-stv with cksource/ckeditor5-commercial#5637

**Review by arkflpc — APPROVED** [review]

**Pull request merged**